pkg/client/package_buildpack.go

```diff
@@ -72,7 +81,12 @@ func (c *Client) PackageBuildpack(ctx context.Context, opts PackageBuildpackOpti
 		return errors.Wrap(err, "creating layer writer factory")
 	}
 
-	packageBuilder := buildpack.NewBuilder(c.imageFactory)
+	var packageBuilderOpts []buildpack.PackageBuilderOption
+	if opts.Flatten {
+		packageBuilderOpts = append(packageBuilderOpts, buildpack.WithFlatten(opts.Depth, opts.FlattenExclude),
+			buildpack.WithLayerWriterFactory(writerFactory), buildpack.WithLogger(c.logger))
```

Comment:
What factory do we fall back to if this option is not provided?

Reply:
The LayerWriterFactory is only used when you enable the flattened mode, so I didn't add a default value. Initially, I wanted to put all three things together: `WithFlatten(deps, exclude, factory, logging)` but then I separated on these 3 options.

Reply:
I see... does `layerWriterFactory` even need to live on the `PackageBuilder` if we know that we always want to use a `*layer.WriterFactory`? Does adding this configuration option help testing?